### ISSUE TITLE
Setting Ruby version was does not work for Homebrew (in `xecute tests on macOS (Xcode 9.4.1, Ruby 2.4)`)

### ISSUE THREAD

**Issue by janpio** [fastlane/fastlane#16536]
See this CI run: https://app.circleci.com/pipelines/github/fastlane/fastlane/318/workflows/5bc3c6da-b4dc-4938-b987-3297b95978a9/jobs/46397/steps

![image](https://user-images.githubusercontent.com/183673/83335573-1e9f7f80-a2ae-11ea-9012-205d1d49df5e.png)
![image](https://user-images.githubusercontent.com/183673/83335579-2b23d800-a2ae-11ea-9afe-9034e476abca.png)

I do not understand why Homebrew tries to use Ruby 2.3 here.

**Comment by janpio** [comment]
Hmm, maybe some cache that is wrong and does not take the version into account? That would also kind of explain why this happens only _sometimes_ I think.

**Comment by janpio** [comment]
Nope, that's not it: https://app.circleci.com/pipelines/github/fastlane/fastlane/319/workflows/0cc8a650-59f6-4250-bb40-d65a3be169b4/jobs/46405/steps

**Comment by janpio** [comment]
Turns out that "Updating Homebrew..." always installs a Ruby 2.6:

Ruby 2.4:
https://app.circleci.com/pipelines/github/fastlane/fastlane/322/workflows/56a94ed8-a0ee-402d-b877-5909f898261a/jobs/46434/steps
```
Untapping caskroom/versions...
 
Untapped 199 casks (360 files, 55.4MB).
 
Updating Homebrew...
 
==> Downloading https://homebrew.bintray.com/bottles-portable-ruby/portable-ruby-2.6.3.mavericks.bottle.tar.gz
 
######################################################################## 100.0%
 
==> Pouring portable-ruby-2.6.3.mavericks.bottle.tar.gz
 
==> Auto-updated Homebrew!
 
Updated 3 taps (homebrew/core, homebrew/cask and adoptopenjdk/openjdk).
 
```

Ruby 2.5:
https://app.circleci.com/pipelines/github/fastlane/fastlane/322/workflows/56a94ed8-a0ee-402d-b877-5909f898261a/jobs/46430/steps
```
Untapping caskroom/versions...
 
Untapped 161 casks (209 files, 317.4KB).
 
Updating Homebrew...
 
==> Downloading https://homebrew.bintray.com/bottles-portable-ruby/portable-ruby-2.6.3.mavericks.bottle.tar.gz
 
######################################################################## 100.0%
 
==> Pouring portable-ruby-2.6.3.mavericks.bottle.tar.gz
 
==> Auto-updated Homebrew!
 
```

**Comment by minuscorp** [comment]
Can we use rvm or rbenv to force the environment to be a certain version of Ruby, even if it adds a little bit of compile time?

**Comment by janpio** [comment]
That's what we are already doing: https://github.com/fastlane/fastlane/blob/fa2ca35ff79c514c9f1e5730ed62fc1aa17361c1/.circleci/config.yml#L56-L61

**Comment by janpio** [comment]
Still no idea what is going on on that image.

Any objections of us just dropping Xcode 9.4.1 now? @joshdholtz
